File: LivrusVirtus/Delphi8.NET/VCL/dbExpress/fontes/ConEditoras.pas
unit ConEditoras;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  Consultas, Db, DBTables, Grids, DBGrids, StdCtrls, ExtCtrls, FMTBcd,
  Borland.Vcl.Provider, Borland.Vcl.DBClient, Borland.Vcl.SqlExpr,
  System.ComponentModel;

type
  TfrmConEditoras = class(TfrmConsultas)
    procedure FormClose(Sender: TObject; var Action: TCloseAction);
    procedure btnLimparClick(Sender: TObject);
    procedure btnPesquisarClick(Sender: TObject);
    procedure FormShow(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  frmConEditoras: TfrmConEditoras;

implementation

uses Principal, ConsultasSQL;

{$R *.DFM}

procedure TfrmConEditoras.FormClose(Sender: TObject;
  var Action: TCloseAction);
begin
frmPrincipal.Editoras2.Enabled:=true;
inherited;
end;

procedure TfrmConEditoras.btnLimparClick(Sender: TObject);
begin
  inherited;
LimparDados(ConsSQL.Editora('S','0',''));
end;

procedure TfrmConEditoras.btnPesquisarClick(Sender: TObject);
begin
  inherited;
PesquisarDados(ConsSQL.Editora('S','', edtDescricao.Text),
'Editora não encontrada!');
end;

procedure TfrmConEditoras.FormShow(Sender: TObject);
begin
  inherited;
btnLimparClick(self);
end;


end.

File: LivrusVirtus/Kylix3/cpp/fontes/ConEditoras.cpp
//---------------------------------------------------------------------------

#include <clx.h>
#pragma hdrstop

#include "ConEditoras.h"
#include "ConsultasSQL.h"
#include "Principal.h"
//---------------------------------------------------------------------------
#pragma link "Consultas"
#pragma resource "*.xfm"
TfrmConEditoras *frmConEditoras;
//---------------------------------------------------------------------------
__fastcall TfrmConEditoras::TfrmConEditoras(TComponent* Owner)
        : TfrmConsultas(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TfrmConEditoras::FormClose(TObject *Sender,
      TCloseAction &Action)
{
frmPrincipal->Editoras2->Enabled=true;
TfrmConsultas::FormClose(Sender, Action);
}
//---------------------------------------------------------------------------
void __fastcall TfrmConEditoras::btnLimparClick(TObject *Sender)
{
LimparDados(ConsSQL->Editora('S',"0",""));
}
//---------------------------------------------------------------------------
void __fastcall TfrmConEditoras::btnPesquisarClick(TObject *Sender)
{
PesquisarDados(ConsSQL->Editora('S',"", edtDescricao->Text.c_bstr()),
"Editora não encontrada!");
}
//---------------------------------------------------------------------------
void __fastcall TfrmConEditoras::FormShow(TObject *Sender)
{
btnLimparClick(this);
}
//---------------------------------------------------------------------------

